csrc/ops/utils.cpp

```diff
@@ -174,12 +174,128 @@ IterType promoteIterType(IterType type1, IterType type2) {
   }
 }
 
+//! Maps the input iterdomains to the output of MatmulOp.
+//! Returns a vector where each element is the
+//! input iterdomain corresponding to the output iterdomain at that index.
```

Comment:
The entries are still the input iterdomain, just aligned to the output.
For eg: Consider `[B, M, K] x [K, N] -> [B, M, N]`. For A: the function returns `{id_B, id_M, nullptr}`. For B: the function returns `{nullptr, nullptr, id_N}`. This allows us to identify which of the input iterdomains correspond to a particular index in the output.

Reply:
The use was to avoid copying this logic into `root_domain_map`. This way, I can use the output of this function to map producer-consumer for `MatmulOp`. Does this clarify the use of the function?

Reply:
We could rename it to `alignMatmulOpInputToOutput` if `mapping` is creating a misunderstanding about the functionality.